lib/l4p/udp_stream.c

```diff
@@ -276,6 +276,74 @@ tle_udp_stream_open(struct tle_ctx *ctx,
 	return &s->s;
 }
 
+	struct tle_stream *
+tle_udp_stream_establish(struct tle_ctx *ctx,
+		const struct tle_udp_stream_param *prm)
+{
+	struct tle_udp_stream *s;
+
+	if (ctx == NULL || prm == NULL || check_stream_prm(ctx, prm) != 0) {
+		rte_errno = EINVAL;
+		return NULL;
+	}
+
+	s = (struct tle_udp_stream *)get_stream(ctx);
+	if (s == NULL)	{
+		rte_errno = ENFILE;
+		return NULL;
+
+		/* some TX still pending for that stream. */
+	} else if (UDP_STREAM_TX_PENDING(s)) {
+		put_stream(ctx, &s->s, 0);
+		rte_errno = EAGAIN;
+		return NULL;
+	}
+
+	/* copy input parameters. */
+	s->prm = *prm;
+
+	/* setup L3 src and dst addresses. */
+	if (prm->local_addr.ss_family == AF_INET) {
+		s->s.type = TLE_V4;
+		s->s.ipv4.addr.dst = ((const struct sockaddr_in *)&prm->local_addr)->sin_addr.s_addr;
+		s->s.ipv4.mask.dst = INADDR_NONE;
+		s->s.ipv4.addr.src = ((const struct sockaddr_in *)&prm->remote_addr)->sin_addr.s_addr;
+		s->s.ipv4.mask.src = INADDR_NONE;
+	} else if (prm->local_addr.ss_family == AF_INET6) {
+		s->s.type = TLE_V6;
+		const struct in6_addr *pm;
+		pm = &tle_ipv6_none;
+		//fill_ipv6_am((const struct sockaddr_in6 *)&prm->local_addr,
```

Comment:
Unnecessary comment?

Reply:
ipv6 has not been tested, this in theory is the correct implementation of that call...  i'd say only unnecessary if the implementation here was actually tested or fully reviewed and found to be accurate.  right now its 50/50 if it works and keeps the addresses in the right order

Reply:
v6 teste, comments removed